service/history/mutableState.go

```diff
@@ -120,37 +120,38 @@ type (
 		GetChildExecutionInfo(int64) (*persistence.ChildExecutionInfo, bool)
 		GetChildExecutionInitiatedEvent(int64) (*workflow.HistoryEvent, bool)
 		GetCompletionEvent() (*workflow.HistoryEvent, bool)
+		GetDecisionInfo(int64) (*decisionInfo, bool)
```

Comment:
The other renaming looks good except for this one. GetDecisionInfo(initiatedID) sounds like it is able to get any decision but it is actually getting current decision. I would prefer GetPendingDecisionInfo or GetCurrentDecisionInfo

Reply:
There is already a GetPendingDecision(). What this does is Get and check if the decisionID is the same.

Reply:
we actually do `GetActivityInfo` which returns activity info and a bool, just following the same pattern here

GetPendingDecisionInfo() basically just get the pending decision, like GetPendingActivityInfos